plugin/core/windows.py

```diff
@@ -437,6 +440,34 @@ def get_project_path(self, file_path: str) -> Optional[str]:
                     candidate = folder
         return candidate
 
+    def should_present_diagnostics(self, uri: DocumentUri) -> Optional[str]:
+        scheme, path = parse_uri(uri)
+        if scheme != "file":
+            return None
+        if not self._workspace.contains(path):
+            return "not inside window folders"
+        view = self._window.active_view()
+        if not view:
+            return None
+        settings = view.settings()
+        if self._matches_glob(path, settings.get("binary_file_patterns")):
+            return "matches a glob in binary_file_patterns"
+        if self._matches_glob(path, settings.get("file_exclude_patterns")):
+            return "matches a glob in file_exclude_patterns"
+        if self._matches_glob(path, settings.get("folder_exclude_patterns")):
+            return "matches a glob in folder_exclude_patterns"
+        return None
+
+    def _matches_glob(self, path: str, patterns: Any) -> bool:
+        if not isinstance(patterns, list):
+            return False
+        for pattern in patterns:
+            if not isinstance(pattern, str):
+                continue
+            if fnmatch.fnmatch(path, pattern):
+                return True
+        return False
```

Comment:
- `fnmatch` is not really a 1:1 implementation of the ST pattern matching. Are we fine with that?
- We could have it as a global utility function since I will need it for file watcher also.
- Those patterns are not "glob"'s technically so the name is a bit misleading.

Reply:
> fnmatch is not really a 1:1 implementation of the ST pattern matching. Are we fine with that?

For now I would say yes. Seeing as we can't really filter `index_exclude_patterns` correctly anyway due to the `index_exclude_gitignore` setting adding a lot of complexity.

> Those patterns are not "glob"'s technically so the name is a bit misleading.

I thought they were... just lacking features.

Reply:
> I thought they were... just lacking features.

I suppose you are right. Though typically `glob` refers to patterns that support `**` but I guess it's not a rule.

Reply:
> > fnmatch is not really a 1:1 implementation of the ST pattern matching. Are we fine with that?
> 
> For now I would say yes. Seeing as we can't really filter `index_exclude_patterns` correctly anyway due to the `index_exclude_gitignore` setting adding a lot of complexity.

But the difference in parsing `/` could make this solution very error with some patterns, potentially.

Reply:
> We could have it as a global utility function since I will need it for file watcher a

Made it into a public function.